arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -538,3 +546,28 @@ public fun <A : AutoCloseable> autoCloseable(
 ): Resource<A> = resource {
   autoCloseable(closingDispatcher, autoCloseable)
 }
+
+public fun CompletableJob.completeWith(exitCase: ExitCase) {
+  when (exitCase) {
+    is ExitCase.Cancelled -> cancel(exitCase.exception)
+    is ExitCase.Failure -> completeExceptionally(exitCase.failure)
+    ExitCase.Completed -> complete()
+  }
+}
+
+public suspend fun CompletableJob.completeWithAndJoin(exitCase: ExitCase) {
+  completeWith(exitCase)
+  return join()
+}
```

Comment:
What is the use case of those functions? I'm a bit worried about making the API too big.